catalog/validator/catalog-validator-metacardwkt/src/main/java/org/codice/ddf/validator/metacard/wkt/MetacardWktValidator.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.validator.metacard.wkt;
+
+import com.google.common.collect.ImmutableSet;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.types.Validation;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ReportingMetacardValidator;
+import ddf.catalog.validation.ValidationException;
+import ddf.catalog.validation.impl.ValidationExceptionImpl;
+import ddf.catalog.validation.impl.report.MetacardValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.MetacardValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.validator.wkt.WktValidator;
+
+public class MetacardWktValidator implements MetacardValidator, ReportingMetacardValidator {
+  private String errorMsg;
+
+  private String validatedAttributeKey;
+
+  private final WktValidator wktValidator;
+
+  public MetacardWktValidator(WktValidator wktValidator, String validatedAttributeKey) {
+    this.wktValidator = wktValidator;
+    this.validatedAttributeKey = validatedAttributeKey;
+    this.errorMsg = validatedAttributeKey + " is invalid: ";
+  }
+
+  @Override
+  public Optional<MetacardValidationReport> validateMetacard(Metacard metacard) {
+    MetacardValidationReportImpl report = null;
+
+    String attributeValue = getAttributeValue(metacard);
+
+    if (StringUtils.isNotEmpty(attributeValue)) {
+      if (!wktValidator.isValid(attributeValue)) {
+        String message = errorMsg + metacard.getLocation();
+        ValidationViolation violation =
+            new ValidationViolationImpl(
+                ImmutableSet.of(validatedAttributeKey),
+                message,
+                ValidationViolation.Severity.ERROR);
+        report = new MetacardValidationReportImpl();
+        report.addMetacardViolation(violation);
+      }
+    } else {
```

Comment:
This is outside the scope of this ticket but it seems like this logic for preserving existing error reports should be contained in the class calling the validators.

Reply:
I agree with you. It should be built into the core of the validation, not in a particular validator. I can make a tech debt ticket.